core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -217,7 +219,8 @@ class ReplicaManager(val config: KafkaConfig,
 
   def startup() {
     // start ISR expiration thread
-    scheduler.schedule("isr-expiration", maybeShrinkIsr, period = config.replicaLagTimeMaxMs, unit = TimeUnit.MILLISECONDS)
+    // A follower can lag behind leader for up to config.replicaLagTimeMaxMs x (1 + 20%) before it is removed from ISR
+    scheduler.schedule("isr-expiration", maybeShrinkIsr, period = config.replicaLagTimeMaxMs / 5, unit = TimeUnit.MILLISECONDS)
```

Comment:
Thanks for the explanation. It makes sense. I am just a bit concerned of the overhead of calling maybeShrinkIsr() to frequently.  Perhaps we can do /2?